pypesto/objective/petab_import.py

```diff
@@ -371,6 +371,10 @@ def __getstate__(self) -> dict:
                 {'amici_model', 'amici_solver', 'edatas'}:
             state[key] = self.__dict__[key]
 
+        amici_solver_file = tempfile.mkstemp()[1]
+        amici.writeSolverSettingsToHDF5(self.amici_solver, amici_solver_file)
+        state['amici_solver'] = amici_solver_file
```

Comment:
is possible to store that file as bytestream in the state dict? In current implementation removing the file will make it impossible to unpickle the same pickle twice and make it really hard to move pickles across machines (need to check amici versions anyways). Nothing of immediate concern, but may become relevant in the future.

Reply:
true. the current implementation in amici requires a file to write to, right? so, currently, one would need to write the file and directly read it in again and save the bytes in the state dict. is that what you mean?